custom/icds_reports/management/commands/quarter_school_data.py

```diff
@@ -44,46 +44,47 @@ def handle(self, *args, **options):
                     ON (
                         child_health_monthly.case_id=ucr.doc_id AND
                         child_health_monthly.supervisor_id=ucr.supervisor_id AND
-                        month>='2020-01-01' AND month<='2020-03-01'
+                        month>='2020-01-01' AND month<'2020-04-01'
                     )
                     WHERE
                         pse_eligible=1 AND
                         age_tranche='{age_group}' AND
                         month>='2020-01-01' AND
-                        month<='2020-03-01';
+                        month<'2020-04-01';
                 """
 
         case_accessor = CaseAccessors('icds-cas')
 
         for age_group in (36, 48, 60, 72):
 
-            cases_ids = {item[1]: item[0] for item in _run_custom_sql_script(query.format(age_group=age_group))}
+            case_id_to_state_id = {item[1]: item[0]
+                                   for item in _run_custom_sql_script(query.format(age_group=age_group))
+                                   }
 
-            for ids_chunk in chunked(list(cases_ids.keys()), 1000000):
+            for ids_chunk in chunked(list(case_id_to_state_id.keys()), 500000):
                 for case in case_accessor.get_cases(list(ids_chunk)):
                     date_return_private = case.get_case_property('date_return_private')
                     if (date_return_private and
                             date(2020, 1, 1) <= parser.parse(date_return_private) < date(2020, 4, 1)):
-                        final_state_data[cases_ids[case.get_case_property('case_id')]][age_group] += 1
+                        final_state_data[case_id_to_state_id[case.case_id]][age_group] += 1
 
         self.write_to_file(final_state_data)
 
     def write_to_file(self, state_json_data):
-        fout = open('/home/cchq/preschool_quarter_data.csv', 'w')
-        writer = csv.writer(fout)
-        state_data_in_list = [
-            'state_name',
-            '3_4',
-            '4_5',
-            '5_6'
-        ]
-        for state_id, state_data in state_json_data.items():
-            state_data_in_list.append([
-                state_data['state_name'],
-                state_data[36] + state_data[48],
-                state_data[60],
-                state_data[72]
-            ])
-
-        writer.writeLines(state_data_in_list)
-        fout.close()
+        with open('/home/cchq/preschool_quarter_data.csv', 'w') as fout:
```

Comment:
I like `/home/cchq` in the file path. I believe you would run this from the current directory but want it to be saved a directory higher instead?

Reply:
yeah, But I dont have any reason to back that over saving it in the local directory. Do you see saving it in the local directory is preferable?

Reply:
offlined. 
An advantage was self clean up with releases and possibly easier scp.